### PR TITLE
Bug 1817070 - Add strings used for experimentation with Re-engagement notifications

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1817070

### CODE REVIEW

**Review by mavduevskiy — APPROVED** [review]
looking good, 🚢 it!

File: fenix/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1177,6 +1177,14 @@
     <!-- Text shown in the notification that pops up to re-engage the user.
     %1$s is a placeholder that will be replaced by the app name. -->
     <string name="notification_re_engagement_text">Browse with no saved cookies or history in %1$s</string>
+    <!-- Title A shown in the notification that pops up to re-engage the user -->
+    <string name="notification_re_engagement_A_title" tools:ignore="UnusedResources">Browse without a trace</string>
+    <!-- Text A shown in the notification that pops up to re-engage the user -->
+    <string name="notification_re_engagement_A_text" tools:ignore="UnusedResources">Private browsing in Firefox doesn’t save your info.</string>
```
[the remark this answers is not in the document]
Comment (rocketsroger) [reply fenix/app/src/main/res/values/strings.xml]: Good catch, didn't notice the "Firefox" there.  Should be the placeholder.

File: fenix/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1177,6 +1177,15 @@
     <!-- Text shown in the notification that pops up to re-engage the user.
     %1$s is a placeholder that will be replaced by the app name. -->
     <string name="notification_re_engagement_text">Browse with no saved cookies or history in %1$s</string>
+    <!-- Title A shown in the notification that pops up to re-engage the user -->
+    <string name="notification_re_engagement_A_title" tools:ignore="UnusedResources">Browse without a trace</string>
+    <!-- Text A shown in the notification that pops up to re-engage the user.
+    %1$s is a placeholder that will be replaced by the app name. -->
+    <string name="notification_re_engagement_A_text" tools:ignore="UnusedResources">Private browsing in %1$s doesn’t save your info.</string>
```
Comment (jhugman) [inline fenix/app/src/main/res/values/strings.xml:1184]: As Nimbus doesn't support substitutions, it'd be useful to have a version of this string with Firefox already embedded in the string.

Comment (Delphine) [reply fenix/app/src/main/res/values/strings.xml:1184]: We should really make sure experiments can use placeholders for app name. Please let me know if there is a reason we can't.
I'm also thinking of Focus/Klar: if we have Nimbus experiments on Focus, we need to support these placeholders.
Regardless, we should use what's in the code. We can't scale doubling strings to accommodate all product names

**Pull request merged**